FiftyOne.DeviceDetection/DeviceDetectionOnPremisePipelineBuilder.cs

```diff
@@ -404,6 +410,111 @@ public DeviceDetectionOnPremisePipelineBuilder SetPerformanceProfile(Performance
+        /// </param>
+        /// <returns>
+        /// This builder instance.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        /// Thrown if url parameter is null or and empty string
+        /// </exception>
+        public DeviceDetectionOnPremisePipelineBuilder SetDataUpdateUrl(string url)
+        {
+            if (string.IsNullOrEmpty(url)) throw new ArgumentNullException(nameof(url));
+            _dataUpdateUrlString = url;
+            _dataUpdateUrlUri = null;
+            return this;
+        }
+
+        /// <summary>
+        /// Configure the engine to use the specified URL when looking for
+        /// an updated data file.
+        /// </summary>
+        /// <param name="url">
+        /// The URL to check for a new data file.
+        /// </param>
+        /// <returns>
+        /// This builder instance.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        /// Thrown if url parameter is null
+        /// </exception>
+        public DeviceDetectionOnPremisePipelineBuilder SetDataUpdateUrl(Uri url)
+        {
+            if (url == null) throw new ArgumentNullException(nameof(url));
+            _dataUpdateUrlString = null;
+            _dataUpdateUrlUri = url;
+            return this;
+        }
+
+        /// <summary>
+        /// Specify a <see cref="IDataUpdateUrlFormatter"/> to be 
+        /// used by the <see cref="DataUpdateService"/> when building the 
+        /// complete URL to query for updated data.
+        /// </summary>
+        /// <param name="formatter">
+        /// The formatter to use.
+        /// </param>
+        /// <returns>
+        /// This builder instance.
+        /// </returns>
+        public DeviceDetectionOnPremisePipelineBuilder SetDataUpdateUrlFormatter(
+            IDataUpdateUrlFormatter formatter)
+        {
+            _dataUpdateUrlFormatter = formatter;
+            _useFormatter = null;
+            return this;
+        }
+
+        /// <summary>
+        /// Enable or disable the <see cref="IDataUpdateUrlFormatter"/>
+        /// to be used when creating the complete URL to request updates
+        /// from.
+        /// </summary>
+        /// <remarks>
+        /// Setting this to false is equivalent to calling 
+        /// <see cref="SetDataUpdateUrlFormatter(IDataUpdateUrlFormatter)"/>
+        /// with a null parameter.
+        /// It is available as a separate method in order to support 
+        /// disabling the formatter from a configuration file.
+        /// </remarks>
+        /// <param name="useFormatter">
+        /// True to use the specified formatter (default). False to
+        /// prevent the specified formatter from being used.
+        /// </param>
+        /// <returns>
+        /// This builder instance.
+        /// </returns>
+        public DeviceDetectionOnPremisePipelineBuilder SetDataUpdateUseUrlFormatter(bool useFormatter)
```

Comment:
let's remove this setter all together, as the presence or absence of urlFormatter should actually decide whether to use it or not - this is how it is in the underlying builder.  the useUrlFormatter is only needed for the JSON configuration, where you can not set urlFormatter to null or object (or can you set it to null?, obviously no way to set an object from json)